src/main/java/fi/okm/jod/yksilo/entity/UriConverter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2025 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.entity;
+
+import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Converter;
+import java.net.URI;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Converter(autoApply = true)
+public class UriConverter implements AttributeConverter<URI, String> {
+  @Override
+  public String convertToDatabaseColumn(URI attribute) {
+    // Serialization: URI object to database string.
+    if (attribute == null) {
+      return null;
+    }
+    if (attribute.getScheme() == null) {
+      throw new IllegalArgumentException("URI scheme is null from: " + attribute);
+    }
+    return attribute.toString();
+  }
+
+  @Override
+  public URI convertToEntityAttribute(String dbData) {
+    // Deserialization: database string to URI object.
+    if (dbData == null) {
+      return null;
+    }
+    try {
+      return URI.create(dbData);
+    } catch (IllegalArgumentException e) {
+      log.error("Failed to convert database value to URI: {}", dbData, e);
+      throw e;
+    }
```

Comment:
Seems unnecessary to add logging here, the exception already conveys the error (and it will be eventually logged).